nilearn/tests/test_niimg.py

```diff
@@ -60,8 +60,11 @@ def test_img_data_dtype():
     dtype_matches = []
     with InTemporaryDirectory():
         for logical_dtype in nifti1_dtypes:
-            dataobj = np.random.uniform(0, 255,
-                                        size=(2, 2, 2)).astype(logical_dtype)
+            dataobj = (
+                np.random.RandomState(42)
```

Comment:
Here also you need to define
`rng = np.random.RandomState(42)`
outside of the loop, then  `call rng.uniform(0, 255, (2, 2, 2)).astype(logical_dtype)` within the loop.